File: core/src/main/java/y2k/joyreactor/Post.java
package y2k.joyreactor;

import java.io.Serializable;
import java.util.Date;

/**
 * Created by y2k on 9/27/15.
 */
public class Post implements Serializable, Comparable<Post> {

    public int id;

    public String title;

    public Image image;

    public UserImage getUserImage() {
        return UserImage.fromUrl(userImage);
    }

    public String userImage;

    public String userName;

    public Date created;
    public String serverId;

    public int commentCount;
    public float rating;

    @Override
    public int compareTo(Post post) {
        return id - post.id;
    }
}

File: core/src/main/kotlin/y2k/joyreactor/Post.kt
package y2k.joyreactor

import y2k.joyreactor.services.repository.DataSet
import java.io.Serializable
import java.util.*

/**
 * Created by y2k on 9/27/15.
 */
class Post : Serializable, Comparable<Post>, DataSet.Dto {

    override var id: Long = 0

    var title: String? = null
    var image: Image? = null
    var userImage: String? = null
    var userName: String? = null
    var created: Date? = null
    var serverId: String? = null
    var commentCount: Int = 0
    var rating: Float = 0.toFloat()

    fun getUserImage(): UserImage {
        return UserImage.fromUrl(userImage)
    }

    override fun compareTo(other: Post): Int {
        return (id - other.id).toInt()
    }
}